### PR TITLE
feat(replays): fetch replay data without projectSlug

### LINKED ISSUE #47863: Use `OrganizationReplayDetails` instead of `ProjectReplayDetails` to fetch data
Follows https://github.com/getsentry/sentry/pull/47859

With the new endpoint we don't need `project` in the replay details url, and can remove it. Once this is removed it'll make constructing links easier in places like discover too.

### PR DESCRIPTION
**Pull request by eliashussary**
## Summary
Adds support for fetching replay details without requiring `projectSlug` in the `replaySlug`. In other words we only need to pass `replayId`. 

With this change we make linking to the replay details page from backend errors and other events simpler.

Relates to: https://github.com/getsentry/sentry/pull/47859
Closes: https://github.com/getsentry/sentry/issues/47863

### CODE REVIEW

File: static/app/utils/replays/hooks/useReplayData.tsx [hunk 1]
```diff
@@ -233,4 +242,38 @@ function useReplayData({
...
+  orgSlug: string,
+  projectSlug: string | null,
+  replayId: string
+) {
+  if (projectSlug) {
+    return `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/`;
+  }
+  return `/organizations/${orgSlug}/replays/${replayId}/`;
+}
+
```
Comment (eliashussary) [inline static/app/utils/replays/hooks/useReplayData.tsx:278]: note to self; reviewing this after the fact... we only need to support the replayId codepath and can ignore projectslug completely

File: static/app/utils/replays/hooks/useReplayData.tsx [hunk 2]
```diff
@@ -233,4 +240,24 @@ function useReplayData({
   };
 }
 
+// see https://github.com/getsentry/sentry/pull/47859
+// replays can apply to many projects when incorporating backend errors
+// this makes having project in the `replaySlug` obsolete
+// we must keep this url schema for now for backward compat but we should remove it at some point
+// TODO: remove support for projectSlug in replay url?
```
Comment (billyvg) [inline static/app/utils/replays/hooks/useReplayData.tsx:247]: Set a calendar invite for 90 days when current replays will be out of retention?

File: static/app/utils/replays/hooks/useReplayData.tsx [hunk 3]
```diff
@@ -103,18 +105,23 @@ function useReplayData({
...
+    const response = await api.requestPromise(makeFetchReplayApiUrl(orgSlug, replayId));
     const mappedRecord = mapResponseToReplayRecord(response.data);
     setReplayRecord(mappedRecord);
     setState(prev => ({...prev, fetchingReplay: false}));
-  }, [api, orgSlug, projectSlug, replayId]);
+  }, [api, orgSlug, replayId]);
 
   const fetchAttachments = useCallback(async () => {
-    if (!replayRecord) {
+    if (!replayRecord || !projectSlug) {
```
Comment (billyvg) [inline static/app/utils/replays/hooks/useReplayData.tsx:124]: Why do we fetch attachments where there is no project slug?

Comment (eliashussary) [reply static/app/utils/replays/hooks/useReplayData.tsx:124]: its not that we were fetching attachments without a project slug, its that before we already had the projectSlug from the URL so we never needed to assert against it.

now the project slug comes after we resolve a replayRecord so we have to wait for that first

Comment (billyvg) [reply static/app/utils/replays/hooks/useReplayData.tsx:124]: Oops sorry, missed the return below this 🤦

**Review by billyvg — APPROVED** [review]
LGTM

**Pull request merged**